Core.Collectors/IO/StorageAccountHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.Core.Collectors.IO
+{
+    public static class StorageAccountHelper
+    {
+        private static readonly string endPointSuffix = "core.windows.net";
+        public static CloudStorageAccount GetStorageAccount(string storageConnectionEnvironmentVariable)
+        {
+            string stagingBlobConnectionString = Environment.GetEnvironmentVariable(storageConnectionEnvironmentVariable);
+            return CloudStorageAccount.Parse(stagingBlobConnectionString);
+        }
+
+        public async static Task<CloudStorageAccount> GetStorageAccountUsingMsi(string storageAccountNameEnvironmentVariable = "StorageAccountName", bool isQueue = false)
```

Comment:
>isQueue [](http://example.com/codeflow?start=154&length=7)

Instead of using this as a parameter and then computing resource inside, do you think we should have the caller pass the "resource" value? That would make things much cleaner on the caller side?